docs/content/docs/connectors/datastream/kinesis.md

```diff
@@ -566,124 +583,124 @@ Retry and backoff parameters can be configured using the `ConsumerConfigConstant
 
-Since Flink 1.4.0, `FlinkKinesisProducer` switches its default underlying KPL from a one-thread-per-request mode to a thread-pool mode. KPL in thread-pool mode uses a queue and thread pool to execute requests to Kinesis. This limits the number of threads that KPL's native process may create, and therefore greatly lowers CPU utilization and improves efficiency. **Thus, We highly recommend Flink users use thread-pool model.** The default thread pool size is `10`. Users can set the pool size in `java.util.Properties` instance with key `ThreadPoolSize`, as shown in the above example.
+### KDS Sinks and Fault Tolerance
 
-Users can still switch back to one-thread-per-request mode by setting a key-value pair of `ThreadingModel` and `PER_REQUEST` in `java.util.Properties`, as shown in the code commented out in above example.
+The sink is designed to participate in Flink's checkpointing to provide at-least-once processing guarantees. It does this by flushing the entire contents of the buffer when a checkpoint reaches the sink. This effectively assures all requests that were triggered before the checkpoint have been successfully acknowledged by Kinesis Data Streams, before proceeding to process more records sent to the sink.
 
-### Backpressure
+In case of a failure or a resharding, data will be written again to Kinesis, leading to duplicates.  Also, the sink does not guarantee that records are written in order to the shards.
 
-By default, `FlinkKinesisProducer` does not backpressure. Instead, records that
-cannot be sent because of the rate restriction of 1 MB per second per shard are
-buffered in an unbounded queue and dropped when their `RecordTtl` expires.
+To use fault tolerant KDS Sinks, checkpointing of the topology needs to be enabled at the execution environment.
 
-To avoid data loss, you can enable backpressuring by restricting the size of the
-internal queue:
+### Backpressure
 
+Backpressure in the sink arises as the sink buffer fills up and writes to the sink 
+begins to exhibit blocking behaviour. Kinesis Data Streams has a rate restriction of 
+1 MB per second per shard.
+
+You can ease backpressuring by adjusting the size of the internal queue:
 ```
-// 200 Bytes per record, 1 shard
-kinesis.setQueueLimit(500);
+KinesisDataStreamsSink<String> kdsSink =
+    KinesisDataStreamsSink.<String>builder()
+        ...
+        .setMaxBufferedRequests(10_000)
+        ...
 ```
 
-The value for `queueLimit` depends on the expected record size. To choose a good
-value, consider that Kinesis is rate-limited to 1MB per second per shard. If
-less than one second's worth of records is buffered, then the queue may not be
-able to operate at full capacity. With the default `RecordMaxBufferedTime` of
-100ms, a queue size of 100kB per shard should be sufficient. The `queueLimit`
-can then be computed via
+The sink default maximum record size is 1MB and maximum batch size is 5MB in line with the Kinesis Data Streams maximums.
```

Comment:
I would keep this line and remove the reference to the default size at line 679-680

Reply:
I was thinking to include both since, the top one (line 679-680) is referring to the flow rate max of 1MB/s/shard which is relevant to debugging customer backpressure. 

But the bottom one refers to a limit of the size of each record submitted, which is 1MB. But I guess this is not too relevant to backpressure, so I've moved it to the main section.